### DISCUSSION TITLE
Using the same vlang compiler installation on windows and linux.

### DISCUSSION THREAD

**Discussion by flysand7** [vlang/v#11889] [Ideas]
Currently my computer has two operating systems installed which I use interchangeably, and both of which access the same filesystem with my projects and other junk.

Yesterday I cloned the repository of vlang compiler  and compiled the compiler from linux. I had used the compiler and it worked fine.

Today I tried to navigate to the same folder and build vlang for windows *also*. Turns out that it didn't do it, because it relied on `tcc.exe` being in `thirdparty\tcc`, and it was there, but it was an ELF executable. The make script, seeing that the tcc compiler is there ignored it's installation and proceeded trying to compile vlang compiler source.

Since windows can not execute ELF executables, I got an error.

I would like for the vlang compiler to execute a different version of tcc (the one for windows or the one for linux) depending on the current OS. For now the solution to my problem is to re-clone vlang and duplicate every single file of the clone library and update and rebuild each one of the copies separately.

**Comment by Gladear** [comment]
To prevent duplicating the whole repository, you can use `git worktree add ../v-windows`
It won't duplicate the whole `.git` directory, which is usually the big part of the repo, and therefore save you some space
